### PR TITLE
Dashboard: Keyboard Navigation for Template Detail View

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Summary
Enables users to use the arrow keys to tab through the detail template pages grid. 

details with keyboard outline

## Relevant Technical Choices

- Took heavily from `edit-story/components/canvas/gridview/useGridViewKeys` to create a custom hook for grid navigation that is not tied to context so that we can reuse it on the different grid cases we have in the dashboard. All I did was update names to be a little more generic since I'll be resusing this for grids containing full stories and just logos rather than only story pages. 
- Also stole from the existing Karma tests to navigate grid with keyboard that exists in `edit-story/components/canvas/karma/gridView.karma.js`

Otherwise, the updates are associated with configuring the card gallery with what the editor did with their grid preview - so buttons as containers and controlling tab index. 

## User-facing changes

Now when you navigate to a template detail page you should be able to use the tab key to reach the preview grid and then once there, use the arrow keys to go right/left/down/up. Hit `Enter` to select a new preview page. Hit `Tab` to exit the grid and go on your merry way. 

## Testing Instructions

Go to template detail page in dashboard, tab 4 times. You should see a slight change in the 'active' border of the smaller grid item that's selected. Now you can use your arrow keys to navigate the grid. When you want to change the page you are actively previewing hit 'enter'. 

---

Addresses #4202

### CODE REVIEW

File: assets/src/dashboard/app/views/shared/storyListView.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ const PreviewContainer = styled.div`
   height: ${({ theme }) => theme.previewWidth.thumbnail / FULLBLEED_RATIO}px;
   vertical-align: middle;
   border-radius: ${({ theme }) => theme.storyPreview.borderRadius}px;
-  border: ${({ theme }) => theme.storyPreview.border};
+  border: ${({ theme }) => theme.borders.gray75};
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/shared/storyListView.js:87]: noticed one border theme setting that wasn't part of the `borders` key, updated the few places it's used to make those more streamlined.

**Comment by carloskelly13** [comment]
This works very well but I'm unsure if the visual indicator is enough. The light blue means you are focused... Perhaps it should become thicker? And we can do that with a box shadow to not affect the overall sizing.

Screen Shot 2020-08-28 at 11 39 45 AM
Screen Shot 2020-08-28 at 11 39 48 AM

File: assets/src/dashboard/components/cardGallery/components.js [hunk 2]
```diff
@@ -27,43 +33,84 @@ export const GalleryContainer = styled.div`
...
+    padding: 0;
+    background-color: transparent;
+    border: ${
+      isSelected ? theme.borders.bluePrimary : theme.borders.transparent
+    };
+    border-width: 2px; 
+
+    ${KEYBOARD_USER_SELECTOR} &:focus {
+      border-radius: 0;
+      border-color: ${rgba(theme.colors.bluePrimary, 0.5)};
```
Comment (carloskelly13) [inline assets/src/dashboard/components/cardGallery/components.js]: Nice use of the polished helpers!

**Review by carloskelly13 — APPROVED** [review]
The functionality and tests work well. 

Two concerns that are minor:

1. The focus indication might not be strong enough
2. While the `useGridViewsKeys` hook has different state logic than the editor's the external functions could be unified. This can be done in a follow up PR as we want to be careful and not introduce any potential spots for regression right now.

**Comment by BrittanyIRL** [comment]
@carlos-kelly  thanks for the suggestions!! 

> 1. The focus indication might not be strong enough

I copied the focus coloring from the edit-story grid preview focus. I agree that it is faint. I will update this to be compliant with AAA standards of contrast. 

> While the useGridViewsKeys hook has different state logic than the editor's the external functions could be unified. This can be done in a follow up PR as we want to be careful and not introduce any potential spots for regression right now.

great call out, I will export the external functions from edit-story and import them into the dashboard's useGridViewsKeys hook to share that functionality. I will also move the unit tests I wrote for those external functions to edit-story to unify the source of truth.

**Comment by BrittanyIRL** [comment]
All updated! 

The opacity for the border color used around the focused item in a grid when using a keyboard is increased to an opacity of `0.85`% which is passing for Graphical Objects and User Interface Components accessibility standards. I'm not sure there's a AAA version of non text color contrast? I can't find one in the w3 docs, let me know if I'm missing something? 

Screen Shot 2020-08-28 at 11 22 03 AM

**Review by dmmulroy — APPROVED** [review]
Good job @BrittanyIRL, looks and works great

**Comment by csossi** [comment]
Verified in QA

**Pull request merged**